libtransmission/tr-udp.cc

```diff
@@ -283,11 +282,9 @@ void tr_session::tr_udp_core::sendto(void const* buf, size_t buflen, struct sock
         // don't warn on bad sockets; the system may not support IPv6
         return;
     }
-    else if (
-        addrport && addrport->address().is_global_unicast_address() &&
-        !session_.global_source_address(tr_af_to_ip_protocol(to->sa_family)))
+    else if (!session_.source_address(tr_af_to_ip_protocol(to->sa_family)))
```

Comment:
This changed needs to be reverted.

Most of the time, loopback addresses like `localhost` and `::1` is reachable via IPv6 despite the machine having no route to any IPv6 destinations. The current code stop Transmission from connecting to loopback addresses even in these cases.